dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/job/TrackerMessageManager.java

```diff
@@ -85,12 +78,9 @@ public void consume( TextMessage message )
         jobConfiguration.setUid( trackerMessage.getUid() );
         trackerImportParams.setJobConfiguration( jobConfiguration );
 
-        TrackerImportThread trackerImportThread = trackerImportThreadFactory.getObject();
-        trackerImportThread.setTrackerImportParams( trackerImportParams );
-
         SecurityContextHolder.getContext()
             .setAuthentication( AuthenticationSerializer.deserialize( trackerMessage.getAuthentication() ) );
 
-        taskExecutor.executeTask( trackerImportThread );
```

Comment:
I think the thread class main purpose was to set the user during asnyc imports. I guess without it the current user is `null` which is equivalent to the `ALL` superuser. Have you tested this? I don't think this is what we would want. The user should be the user who triggered the import.

Reply:
mmm... maybe this is not a problem in this particular case as the importer is considering the user running it the one present in the params and it is validating all ownerships against it, but it could potentially still be a security issue if we can get some metadata that we shouldn't have access to.

Now my doubt is, is it safe that AsyncTaskExecutor is not carrying the SecurityContext? Even more that we assume that a `null` authentication object is equivalent to `ALL` superuser?